File: src/main/java/leetcode/hot100/LeetCode448.java
package leetcode.hot100;

import java.util.ArrayList;
import java.util.List;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/19 14:09
 */
public class LeetCode448 {

    public List<Integer> findDisappearedNumbers(int[] nums) {
        if (nums == null || nums.length == 0) {
            return new ArrayList<>();
        }

        for (int n : nums) {
            if (nums[Math.abs(n) - 1] > 0) {
                nums[Math.abs(n) - 1] = -1 * nums[Math.abs(n) - 1];
            }
        }

        List<Integer> res = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                res.add(i + 1);
            }
        }
        return res;
    }
}

File: src/main/go/leetcode/lists/hot100/LeetCode448.go
package hot100

func findDisappearedNumbers(nums []int) []int {
	if len(nums) == 0 {
		return []int{}
	}
	for _, n := range nums {
		if n < 0 {
			n = n * -1
		}
		index := n - 1
		if nums[index] > 0 {
			nums[index] = nums[index] * -1
		}
	}
	var res []int
	for i := 0; i < len(nums); i++ {
		if nums[i] > 0 {
			res = append(res, i+1)
		}
	}
	return res
}
